src/app/files/file-explorer.js

```diff
@@ -411,14 +411,38 @@ fileExplorer.prototype.toGist = function (id) {
           token: tokenAccess
         })
         if (id) {
+          const fileList = Object.keys(this.files.origGistFiles)
+          const updatedFileList = Object.keys(packaged)
+          // Telling the GIST API to remove files
+          const allItems = fileList
+            .filter(fileName => updatedFileList.indexOf(fileName) === -1)
+            .reduce((acc, deleteFileName) => ({
+              ...acc,
+              [deleteFileName]: null
+            }), this.files.origGistFiles)
+          // adding new files
+          updatedFileList.forEach((el) => {
+            allItems[el] = packaged[el]
+          })
           tooltip('Saving gist (' + id + ') ...')
           gists.edit({
             description: description,
             public: true,
-            files: packaged,
+            files: allItems,
             id: id
           }, (error, result) => {
             proccedResult(error, result)
+            if (!error) {
+              let finalAllItemsArray = Object.keys(allItems)
+              // updating the origGistFiles with the last one saved to the GIST
+              finalAllItemsArray = finalAllItemsArray.filter(fileName => allItems[fileName] !== null)
```

Comment:
It's good practice to pipe your array methods to avoid using let. Also if possible use more explicit variable names. It's hard for me to understand what it's doing.
```typescript
const goodVariableName = Object
  .keys(allItems)
  .reduce((acc, fileName) => ({ ...acc, [fileName] = allItems[fileName] }), {});
```
But for me this is exactly equal to `this.files.origGistFiles = {...allItems};`